identity/app/controllers/EditProfileController.scala

```diff
@@ -47,53 +51,84 @@ class EditProfileController(idUrlBuilder: IdentityUrlBuilder,
   def displayDigitalPackForm: Action[AnyContent] = displayForm(digitalPackPage)
   def displayPrivacyForm: Action[AnyContent] = displayForm(privacyPage)
 
-  private def displayForm(page: IdentityPage) = csrfAddToken {
-    recentlyAuthenticated.async { implicit request =>
-      Future(profileFormsView(page = page, forms = ProfileForms(request.user, PublicEditProfilePage), request.user))
-    }
-  }
-
   def submitPublicProfileForm(): Action[AnyContent] = submitForm(publicPage)
   def submitAccountForm(): Action[AnyContent] = submitForm(accountPage)
   def submitPrivacyForm(): Action[AnyContent] = submitForm(privacyPage)
 
-  def identifyActiveSubmittedForm(page: IdentityPage): EditProfilePage =
-    page match {
-      case `publicPage` => PublicEditProfilePage
-      case `accountPage` => AccountEditProfilePage
-      case _ => PrivacyEditProfilePage
+  private def displayForm(page: IdentityPage) = csrfAddToken {
+    recentlyAuthenticated.async { implicit request =>
+      Future {
+        profileFormsView(
+          page = page,
+          forms = ProfileForms(request.user, PublicEditProfilePage),
+          request.user)
+      }
     }
+  }
 
-  def submitForm(page: IdentityPage): Action[AnyContent] = csrfCheck { authActionWithUser.async { implicit request =>
-      val activePage = identifyActiveSubmittedForm(page)
-      val user = request.user
-      val forms = ProfileForms(user, activePage).bindFromRequest(request)
-
-      forms.activeForm.value.map {
-        case data: AccountFormData if (data.deleteTelephone) => {
-          identityApiClient.deleteTelephone(user.auth) map {
-            case Left(errors) => profileFormsView(page, forms.withErrors(errors), user)
+  private def submitForm(page: IdentityPage): Action[AnyContent] =
+    csrfCheck {
+      authActionWithUser.async { implicit request =>
+        def identityPageToEditProfilePage(identityPage: IdentityPage): EditProfilePage =
```

Comment:
This method seems a little out of place.  The question I have is, do we even need it?  Why can't `EditProfilePage` inherit from `IdentityPage` and its children define the values of `IdentityPage` fields.  E.g.  
```
sealed trait EditProfilePage extends IdentityPage
case object AccountEditProfilePage extends EditProfilePage("/public/edit", "Edit Public Profile")
```

and then there's no need for a mapping?